File: client/python/mesh/system/mesh_datahouse.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import Any, List

from mesh.kinds import Document, Paging, Page
from mesh.macro import spi
from mesh.mpc import ServiceProxy
from mesh.prsim import DataHouse


@spi("mesh")
class MeshDataHouse(DataHouse):

    def __init__(self):
        self.proxy = ServiceProxy.default_proxy(DataHouse)

    def writes(self, docs: List[Document]):
        return self.proxy.writes(docs)

    def write(self, doc: Document):
        return self.proxy.write(doc)

    def read(self, index: Paging) -> Page[Any]:
        return self.proxy.read(index)

    def indies(self, index: Paging) -> Page[Any]:
        return self.proxy.indies(index)

    def tables(self, index: Paging) -> Page[Any]:
        return self.proxy.tables(index)

File: client/java/src/main/java/io/be/mesh/system/MeshDataHouse.java
package io.be.mesh.system;

import io.be.mesh.macro.SPI;
import io.be.mesh.mpc.ServiceProxy;
import io.be.mesh.prsim.DataHouse;
import io.be.mesh.struct.Document;
import io.be.mesh.struct.Page;
import io.be.mesh.struct.Paging;
import org.springframework.util.CollectionUtils;

import java.util.List;

/**
 * @author jianyue.li
 * @date 2022/3/14 7:47 PM
 */
@SPI("mesh")
public class MeshDataHouse implements DataHouse {

    private final DataHouse dataHouse = ServiceProxy.proxy(DataHouse.class);

    @Override
    public void writes(List<Document> docs) {
        if (!CollectionUtils.isEmpty(docs)) {
            dataHouse.writes(docs);
        }
    }

    @Override
    public void write(Document doc) {
        if (doc != null) {
            dataHouse.write(doc);
        }
    }

    @Override
    public Page<Object> read(Paging index) {
        return dataHouse.read(index);
    }

    @Override
    public Page<Object> indies(Paging index) {
        return dataHouse.indies(index);
    }

    @Override
    public Page<Object> tables(Paging index) {
        return dataHouse.tables(index);
    }

}
